src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -18,82 +18,23 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
+        internal sealed class Boxed
         {
-            private TypeSymbol _defaultType;
-            private int _nullableAnnotation;
-            private Extensions _extensions;
-
-            /// <summary>
-            /// The underlying type, unless overridden by _extensions.
-            /// </summary>
-            internal TypeSymbol DefaultType => Volatile.Read(ref _defaultType);
-
-            /// <summary>
-            /// True if the fields of the builder are unset.
-            /// </summary>
-            internal bool IsDefault => Volatile.Read(ref _extensions) == null;
-
-            /// <summary>
-            /// Set the fields of the builder.
-            /// </summary>
-            /// <remarks>
-            /// This method guarantees: fields will be set once; exactly one caller is
-            /// returned true; and IsDefault will return true until all fields are initialized.
-            /// This method does not guarantee that all fields will be set by the same
-            /// caller. Instead, the expectation is that all callers will attempt to initialize
-            /// the builder with equivalent TypeWithAnnotations instances where
-            /// different fields of the builder may be assigned from different instances.
-            /// </remarks>
-            internal bool InterlockedInitialize(TypeWithAnnotations type)
-            {
-                if (!IsDefault)
-                {
-                    return false;
-                }
-
-                Interlocked.CompareExchange(ref _nullableAnnotation, (int)type.NullableAnnotation, 0);
-                Interlocked.CompareExchange(ref _defaultType, type.DefaultType, null);
-
-                // Because _extensions always gets a non-null value when the struct is initialized, we use it
-                // as a flag to signal that the initialization is complete. This means _extensions should be
-                // initialized last.
-                bool wasFirst = Interlocked.CompareExchange(ref _extensions, type._extensions ?? Extensions.Default, null) is null;
-
-                Debug.Assert(_extensions == (type._extensions ?? Extensions.Default));
-                Debug.Assert(_nullableAnnotation == (int)type.NullableAnnotation);
-                Debug.Assert(_defaultType.Equals(type.DefaultType));
-                return wasFirst;
-            }
+            internal readonly TypeWithAnnotations Value;
 
-            /// <summary>
-            /// We should not be adding any new usages of this method. The only one is currently in SourcePropertySymbol,
-            /// which currently sets the property's type twice in error scenarios.
-            /// We should be able to remove this method by fixing https://github.com/dotnet/roslyn/issues/35381
-            /// </summary>
-            [EditorBrowsable(EditorBrowsableState.Never)]
-            internal void InterlockedDangerousReset()
+            private Boxed(TypeWithAnnotations value)
```

Comment:
>private Boxed(TypeWithAnnotations value) [](start = 12, length = 40)

I do not like this change. One should be able to create instances of this type and to whatever with them